contracts/src/tests/test_access_controller.cairo

```diff
@@ -19,3 +28,58 @@ fn test_upgrade_not_owner() {
 
     AccessController::upgrade(class_hash_const::<2>());
 }
+
+#[test]
+#[available_gas(2000000)]
+fn test_access_control() {
+    let owner = setup();
+    // Deploy access controller
+    let mut calldata = ArrayTrait::new();
+    calldata.append(owner.into()); // owner
+    let (accessControllerAddr, _) = deploy_syscall(
+        AccessController::TEST_CLASS_HASH.try_into().unwrap(), 0, calldata.span(), false
+    ).unwrap();
+
+    should_implement_access_control(accessControllerAddr, owner);
+}
+
+//
+// Tests for contracts that inherit AccessControl.
+// Write functions are assumed to be protected by Ownable::only_owner,
+// but this test does not check for that.
+//
+
+#[abi]
+trait IAccessController {
+    fn has_access(user: ContractAddress, data: Array<felt252>) -> bool;
+    fn add_access(user: ContractAddress);
+    fn remove_access(user: ContractAddress);
+    fn enable_access_check();
+    fn disable_access_check();
+}
+
+fn should_implement_access_control(contract_addr: ContractAddress, owner: ContractAddress) {
+    let contract = IAccessControllerDispatcher { contract_address: contract_addr };
+    let acc2: ContractAddress = contract_address_const::<2222987765>();
+
+    set_contract_address(owner); // required to call contract as owner
```

Comment:
can you elaborate what this is used for? I thought it was so you can set the value of starknet::info::get_contract_address() but its use seems to be very different.

Reply:
Nvm I found out why[ here](https://discord.com/channels/793094838509764618/1065544063245365288/1105507121749049404): 

> you can view your test as a contract that calls another contract. If you want the get_caller_address() in your remote contract to be mocked, then you will need to use set_contract_address() in the test